legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/project/ProjectStructure.java

```diff
@@ -725,6 +686,63 @@ else if (!Arrays.equals(currentBytes, newBytes))
         return projectFileAccessProvider.getFileModificationContext(projectId, workspaceId, workspaceType, workspaceAccessType, revisionId).submit(updateBuilder.getMessage(), operations);
     }
 
+    private static void updateOrAddDependencies(Set<ProjectDependency> toUpdateProjectDependencies, ProjectFileAccessProvider projectFileAccessProvider, Set<ProjectDependency> projectDependencies, boolean purgeOldDependency)
+    {
+        List<ProjectDependency> unknownDependencies = Lists.mutable.empty();
+        List<ProjectDependency> nonProdDependencies = Lists.mutable.empty();
+        SortedMap<ProjectDependency, Exception> accessExceptions = new TreeMap<>();
+        for (ProjectDependency projectDependency : toUpdateProjectDependencies)
+        {
+            try
+            {
+                ProjectConfiguration dependencyConfig = getProjectConfiguration(projectFileAccessProvider.getFileAccessContext(projectDependency.getProjectId(), projectDependency.getVersionId()));
```

Comment:
What if projectDependency has a new-style project id (i.e., groupId:artifactId)? Won't this fail?